designdocs/passkey_p256.md

```diff
@@ -0,0 +1,556 @@
+    - [Web Wallet backend](#web-wallet-backend)
+    - [Users Device](#users-device)
+  - [5. Specification](#5-specification)
+    - [Terms](#terms)
+    - [Passkey Transaction Payload](#passkey-transaction-payload)
+    - [Passkey Registration](#passkey-registration)
+    - [Transaction Submission Specification](#transaction-submission-specification)
+    - [Security Considerations](#security-considerations)
+      - [Generic](#generic)
+      - [Front-end (client)](#front-end-client)
+      - [Backend](#backend)
+      - [On-chain](#on-chain)
+  - [6. Implementation](#6-implementation)
+    - [On-chain Implementation](#on-chain-implementation)
+      - [Extrinsic verification](#extrinsic-verification)
+      - [Signature verification](#signature-verification)
+    - [Backend Example Code](#backend-example-code)
+      - [Passkey Registration Verification](#passkey-registration-verification)
+      - [Transaction Authentication Verification](#transaction-authentication-verification)
+  - [7. Options for Discussion](#7-options-for-discussion)
+    - [Mortality of `account_sig_passkey_pk` i.e. Account Signature on Passkey Public Key](#mortality-of-account_sig_passkey_pk-ie-account-signature-on-passkey-public-key)
+      - [Pros/Cons](#proscons)
+    - [Unsigned Extensions vs Extending MultiSignature](#unsigned-extensions-vs-extending-multisignature)
+      - [Unsigned Extensions](#unsigned-extensions)
+        - [Pros/Cons](#proscons-1)
+      - [Extending MultiSignature](#extending-multisignature)
+        - [Pros/Cons](#proscons-2)
+    - [Generic Key Support](#generic-key-support)
+    - [Separate Pallet](#separate-pallet)
+  - [8. References](#8-references)
+
+## 1. Introduction
+
+This document outlines the design considerations and specifications for integrating P256 Passkey support for performing transactions on Frequency chain. Passkey support aims to provide a novel non custodial solution for managing user accounts and signing transactions on-chain.
+
+## 2. Benefits
+
+- **Non-Custodial Account Management**: Passkey support enables users to interact with Frequency chain without actual seed access. This reduces the risk of seed exposure and enhances security.
+
+- **Functionality**: Passkey support provides a better user experience by enabling users to sign transactions without seed access. The poliferation of passkey enabled services can lead to a seamless user experience.
+
+- **Interoperability**: The use of standard cryptographic protocols (e.g., P256) ensures compatibility with other systems and services that also support these standards. This can facilitate integrations with existing wallets, security modules, and third-party applications.
+
+- **Backup and Recovery**: Passkey support can be used to facilitate account recovery and backup mechanisms. By signing the account public key, users can prove ownership of the account and recover access in case of loss of seed access.
```

Comment:
Perhaps:

```suggestion
- **Backup and Recovery**: Passkey support can be used to facilitate account recovery and backup mechanisms. By signing the account public key, users can prove ownership of the account and recover access in case of a lost seed.
```